### DISCUSSION TITLE
Supavisor support for prisma w/out pgbouncer mode

### DISCUSSION THREAD

**Discussion by heyskylark** [supabase/supabase#22811] [Questions]
Hello,

I'm currently running into severe database performance issues. I located that the problem is connected to Prisma's pgbouncer mode which effects batch processing, such as insert many: https://github.com/prisma/prisma/issues/16081#issuecomment-1511522206

Our current setup is on Vercel, and we're connecting to supavisor through a Prisma client.  One of the solutions that we were given, is that we can remove the Prisma pgbouncer mode from the URI string, but this solution also required turning on prepared statements on pgbouncer: https://github.com/prisma/prisma/issues/16081#issuecomment-1768094738

I was wondering, is a similar solution available for the supavisor proxy?

Thanks!

**Comment by GaryAustin1** [comment]
Just FYI. https://github.com/supabase/supavisor/issues/239

**Comment by heyskylark** [comment]
Hey all, I wanted to update on this issue. I managed to find a work around by using `$executeRaw` and `$queryRaw` and manually writing my own batch inserts. This isn't ideal but in the mean time it works! I was wondering if there's any plans to fix this issue?

For example, I did a few tests with `pgbouncer` flag enabled on the URI and these were my results:

### Batch insert 8,000 records (no nested creates, just creating the one row for each record)
- **createMany query time**: 248,270ms Roughly more than 4 min to complete.
- **$executeRaw query time**: Between 900ms - 1500 ms

### Batch insert 5 records
Even with small batches, there is a significant difference in performance
 - **createMany query time**: 257ms
- **$executeRaw query time**: 56ms
